util/testutil/shell.go

```diff
@@ -141,6 +143,46 @@ func (m *RemoteSnapshotMonitor) CheckAllRemoteSnapshots(t *testing.T) {
 	}
 }
 
+// FatalMonitor scans logs streamed from the specified io.Reader for fatal errors during startup
+func FatalMonitor(r shell.Reporter, stdout, stderr io.Reader) {
+	errs := make(chan error, 1)
+	go scanFatalLog(io.TeeReader(stdout, r.Stdout()), errs)
+	go scanFatalLog(io.TeeReader(stderr, r.Stderr()), errs)
+	select {
+	case err := <-errs:
+		if err != nil {
+			r.Errorf(err.Error())
+		}
+	case <-time.After(10 * time.Second): // timeout
+		r.Errorf("log did not produce success or fatal error within 10 seconds")
+	}
+}
+
+type LevelLogLine struct {
+	Level string `json:"level"`
+	Msg   string `json:"msg"`
+}
+
+// scanFatalLog scans the log streamed from the specified io.Reader for fatal errors during startup
+func scanFatalLog(inputR io.Reader, errs chan error) {
+	scanner := bufio.NewScanner(inputR)
+	var logline LevelLogLine
+	for scanner.Scan() {
+		rawL := scanner.Text()
+		if i := strings.Index(rawL, "{"); i > 0 {
+			rawL = rawL[i:] // trim garbage chars; expects "{...}"-styled JSON log
+		}
+		if err := json.Unmarshal([]byte(rawL), &logline); err == nil {
+			if logline.Level == "fatal" {
+				errs <- errors.New("aborting test: fatal snapshotter log entry encountered")
+			}
+			if strings.Contains(logline.Msg, "background") {
```

Comment:
Noting that we already have an issue to make this more robust: https://github.com/awslabs/soci-snapshotter/issues/515#issuecomment-1479956859